src/script/interpreter.cpp

```diff
@@ -1569,21 +1568,20 @@ static bool VerifyWitnessProgram(const CScriptWitness& witness, int witversion,
             if (stack.size() == 0) {
                 return set_error(serror, SCRIPT_ERR_WITNESS_PROGRAM_WITNESS_EMPTY);
             }
-            const valtype& script_bytes = SpanPopBack(stack);
-            scriptPubKey = CScript(script_bytes.begin(), script_bytes.end());
+            const valtype& script = SpanPopBack(stack);
```

Comment:
This would be much less confusing to read if reordered:
```c++
uint256 hashScriptPubKey;
CSHA256().Write(stack.back().data(), stack.back().size()).Finalize(hashScriptPubKey.begin());
if (memcmp(hashScriptPubKey.begin(), program.data(), 32)) {
    return set_error(serror, SCRIPT_ERR_WITNESS_PROGRAM_MISMATCH);
}
return ExecuteWitnessScript(stack, SpanPopBack(stack), flags, SigVersion::WITNESS_V0, checker, serror);
```
Instead of as-is, popping off the stack just before it would've been useful.

Reply:
Huh, this code just didn't make sense - it had `script` already, why was it accessing it through `witness.stack.back()`?

I've changed it to just use `Write(script.data(), script.size())`.